### ISSUE TITLE
Support license checking and in-app purchases without the Google Play Store

### ISSUE THREAD

**Issue by ghost** [microg/GmsCore#2495]
**Is your feature request related to a problem? Please describe.**
In order to get license checking and in-app purchases to work you need the Google Play Store app. It would be good if microG didn't depend on the Google Play Store app.

**Describe the solution you'd like**
License checking and in-app purchases without the Google Play Store.

**Comment by ale5000-git** [comment]
microG already support license verification but you must update both microG Services and microG Companion and enable the options in microG settings.
It may not work with all apps but it already work in many cases.

**Comment by ale5000-git** [comment]
No, microG Companion (new name of FakeStore) is the replacement of Google Play Store.
It doesn't allow to download apps but it support some things like license verification (it weren't supported at the start but now it is).

**Note:** It must be enabled in microG settings otherwise it doesn't work.

**Comment by CoelacanthusHex** [comment]
> microG already support license verification but you must update both microG Services and microG Companion and enable the corresponding options in microG settings. It may not work with all apps but it already work in many cases.

Noteworthy, it doesn't work with the in-app purchase ordered not in the current device.

https://github.com/microg/GmsCore/pull/2321#issuecomment-2082174738

**Comment by marbens-arch** [comment]
Shouldn't this issue be closed now? MicroG answers license verification requests, and now supports handling Google Play billing requests (including in-app purchases), without the Google Play store, in the current preview release.

**Comment by CoelacanthusHex** [comment]
> Shouldn't this issue be closed now? MicroG answers license verification requests, and now supports handling Google Play billing requests (including in-app purchases), without the Google Play store, in the current preview release.

No, it doesn't work with the in-app purchase ordered not in the current device.

**Comment by n1mbytheone** [comment]
I'd like to add a concrete data point to this issue.
I purchased an in-app product (Noteshelf 3 Premium) on a different Android device using the official Google Play Store, with the same Google account that is configured in microG on my primary device (Volla Phone / Volla OS).
On the microG device, the purchase is not recognized. The app reports "No Products Found", and the underlying request fails with:
Network request failed (POST https://play-fe.googleapis.com/fdfe/skuDetails) invalid: 400
This matches what was mentioned earlier in this thread — that purchases made on a different device than the one running microG are not currently recognized. I just wanted to confirm this is still the case on a recent microG build, and ask if there's a workaround or if this is on the roadmap.
Device: Volla Phone, Volla OS 15.0 20260601-STABLE-mimir

microG version: 0.3.15.250932

App: Noteshelf 3 (Fluid Touch)

**Comment by Eridyne** [comment]
I can also confirm this. I have several apps with old in-app purchases, all made under the Google accounts that are currently configured in microG, and I'm seeing what appears to be the same behavior, with the latest microG as of this post. 
OnePlus 9 Pro, CrDroid 12.11

microG self-test is all good, and all options were enabled for testing.

This seems like a meaningful barrier to microG adoption.
